instance/tests/integration/integration_instance.py

```diff
@@ -489,12 +490,22 @@ def test_openstack_server_terminated(self):
         """
         Test that OpenStackServer detects if the VM was terminated externally.
         """
-        server = OpenStackServer(name_prefix="integration_test")
-        server.save()
-        server.start()
-        server.sleep_until(lambda: server.status.accepts_ssh_commands, timeout=120)
-        server.os_server.delete()
-        self.assert_server_terminated(server)
+        retries = 1
+        while retries >= 0:
+            server = OpenStackServer(name_prefix="integration_test")
+            server.save()
+            server.start()
+            try:
+                server.sleep_until(lambda: server.status.accepts_ssh_commands, timeout=120)
+            except TimeoutError:
+                server.os_server.delete()
+                retries -= 1
+            else:
```

Comment:
I'd like to take the `else` part out of the try/except block to make it clear that verifying that the server had been terminated is the main part of the test.